### PR TITLE
New MGMT API resource for action to forced switch.

### PR DESCRIPTION
**Pull request by kaizimmerm**
Removed `final` modifier from interfaces as well.

Reviewers: @schabdo @melanieretter 

Signed-off-by: kaizimmerm <kai.zimmermann@bosch-si.com>

### CODE REVIEW

File: hawkbit-mgmt-resource/src/main/java/org/eclipse/hawkbit/mgmt/rest/resource/MgmtTargetMapper.java [hunk 1]
```diff
@@ -194,13 +201,32 @@ static MgmtAction toResponse(final String targetId, final Action action, final b
         return result;
     }
 
+    static MgmtAction toResponseWithLinks(final String controllerId, final Action action) {
+        final MgmtAction result = toResponse(controllerId, action);
+
+        if (!action.isCancelingOrCanceled()) {
```
Comment (schabdo) [inline hawkbit-mgmt-resource/src/main/java/org/eclipse/hawkbit/mgmt/rest/resource/MgmtTargetMapper.java]: Use a simple if-else statement. It is better readable and simplifies the code:
```
if (action.isCancelingOrCanceled()){
   result.add(linkTo(methodOn(MgmtTargetRestApi.class).getAction(controllerId, action.getId()))
             .withRel(MgmtRestConstants.TARGET_V1_CANCELED_ACTION));
} else {
 result.add(linkTo(methodOn(MgmtDistributionSetRestApi.class).
     getDistributionSet(action.getDistributionSet().getId())).withRel("distributionset"));
}
```

File: hawkbit-mgmt-resource/src/main/java/org/eclipse/hawkbit/mgmt/rest/resource/MgmtTargetResource.java [hunk 2]
```diff
@@ -302,4 +282,24 @@ private Target findTargetWithExceptionIfNotFound(final String controllerId) {
...
 
+    @Override
+    public ResponseEntity<MgmtAction> updateAction(@PathVariable("controllerId") final String controllerId,
+            @PathVariable("actionId") final Long actionId, @RequestBody final MgmtActionRequestBodyPut actionUpdate) {
+
+        Action action = deploymentManagement.findAction(actionId)
+                .orElseThrow(() -> new EntityNotFoundException(Action.class, actionId));
+        if (!action.getTarget().getControllerId().equals(controllerId)) {
+            LOG.warn("given action ({}) is not assigned to given target ({}).", action.getId(), controllerId);
+            return new ResponseEntity<>(HttpStatus.NOT_FOUND);
```
Comment (schabdo) [inline hawkbit-mgmt-resource/src/main/java/org/eclipse/hawkbit/mgmt/rest/resource/MgmtTargetResource.java]: Just a hint: There is also a builder available => `ResponseEntity.notFound().build()`. But this is totally fine as well

File: hawkbit-mgmt-resource/src/main/java/org/eclipse/hawkbit/mgmt/rest/resource/MgmtRestModelMapper.java [hunk 3]
```diff
@@ -75,4 +68,28 @@ public static ActionType convertActionType(final MgmtActionType actionTypeRest)
             throw new IllegalStateException("Action Type is not supported");
         }
     }
+
+    /**
+     * Convert a action repository type to rest type.
```
Comment (melanieretter) [inline hawkbit-mgmt-resource/src/main/java/org/eclipse/hawkbit/mgmt/rest/resource/MgmtRestModelMapper.java]: an action

File: hawkbit-mgmt-resource/src/test/java/org/eclipse/hawkbit/mgmt/rest/resource/MgmtTargetResourceTest.java [hunk 4]
```diff
@@ -1074,12 +1076,41 @@ public void getUpdateAction() throws Exception {
...
                 .andExpect(jsonPath("_links.distributionset.href",
                         equalTo("http://localhost/rest/v1/distributionsets/"
                                 + actions.get(1).getDistributionSet().getId())))
                 .andExpect(jsonPath("_links.status.href",
-                        equalTo(generateStatusreferenceLink(knownTargetId, actions.get(1)))));
+                        equalTo(generateStatusreferenceLink(knownTargetId, actions.get(1).getId()))));
+    }
+
+    @Test
+    @Description("Verfies that an action is switched from soft to forced if requested by management API")
```
Comment (melanieretter) [inline hawkbit-mgmt-resource/src/test/java/org/eclipse/hawkbit/mgmt/rest/resource/MgmtTargetResourceTest.java]: Verifies

**Pull request merged**